### PR TITLE
SparklerConfig upgrade

### PR DESCRIPTION
**Pull request by SHASHANK-PRO-05**
#93 
This task is still in progress. The bean object has been created but testing needs to be done and once that is done we can start injecting the bean object into the system.
@thammegowda, @karanjeets  please review it.

### CODE REVIEW

**Comment by thammegowda** [comment]
@SHASHANK-PRO-05 Thanks for doing this.  We will take a few days to thoroughly test this.

It looks neat to me.

@karanjeets Check this out. The config upgrade is here.

**Comment by thammegowda** [comment]
@SHASHANK-PRO-05 You have been invited to join the @USCDataScience/sparkler-committer team!
Here is a plan:
We both shall work on `SPARKLER-93` branch (you now have write access to it).
Goal is to completely replace older JSON based config with your newer awesome YAML config.

Then we merge `SPARKLER-93` to develop.
One more round of testing.
Then we merge develop to master.

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/BaseConfig.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package edu.usc.irds.sparkler;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/BaseConfig.java]: Needs ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/SparklerConfig.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
+package edu.usc.irds.sparkler;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/SparklerConfig.java]: needs ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/CrawldbProps.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/CrawldbProps.java]: ALV2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/FetcherProps.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/FetcherProps.java]: ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/GenerateProps.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/GenerateProps.java]: ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/Headers.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/Headers.java]: ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/KafkaProps.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/KafkaProps.java]: ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/PluginsProps.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/PluginsProps.java]: ALv2 header

File: sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/SparkProps.java [hunk 9]
```diff
@@ -0,0 +1,17 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-api/src/main/java/edu/usc/irds/sparkler/configUtils/SparkProps.java]: ALv2 header

File: sparkler-api/src/test/java/edu/usc/irds/sparkler/SparklerConfigTest.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package edu.usc.irds.sparkler;
```
Comment (chrismattmann) [inline sparkler-api/src/test/java/edu/usc/irds/sparkler/SparklerConfigTest.java]: ALv2 header

File: sparkler-plugins/fetcher-jbrowser/src/main/java/edu/usc/irds/sparkler/configUtils/FetcherJBrowserProps.java [hunk 11]
```diff
@@ -0,0 +1,82 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-plugins/fetcher-jbrowser/src/main/java/edu/usc/irds/sparkler/configUtils/FetcherJBrowserProps.java]: ALv2 header

File: sparkler-plugins/fetcher-jbrowser/src/test/java/edu/usc/irds/sparkler/configUtils/FetcherJBrowserConfigTest.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-plugins/fetcher-jbrowser/src/test/java/edu/usc/irds/sparkler/configUtils/FetcherJBrowserConfigTest.java]: ALv2 header

File: sparkler-plugins/urlfilter-regex/src/main/java/edu/usc/irds/sparkler/configUtils/UrlFilterProps.java [hunk 13]
```diff
@@ -0,0 +1,70 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-plugins/urlfilter-regex/src/main/java/edu/usc/irds/sparkler/configUtils/UrlFilterProps.java]: ALv2 header

File: sparkler-plugins/urlfilter-regex/src/test/java/edu/usc/irds/sparkler/configUtils/UrlFilterPropsTest.java [hunk 14]
```diff
@@ -0,0 +1,16 @@
+package edu.usc.irds.sparkler.configUtils;
```
Comment (chrismattmann) [inline sparkler-plugins/urlfilter-regex/src/test/java/edu/usc/irds/sparkler/configUtils/UrlFilterPropsTest.java]: ALv2 header

**Comment by chrismattmann** [comment]
ping @SHASHANK-PRO-05 @thammegowda please see my review

**Comment by SHASHANK-PRO-05** [comment]
@chrismattmann added ALv2 headers.

**Comment by chrismattmann** [comment]
awesome is this ready to review again @SHASHANK-PRO-05 ?

**Comment by SHASHANK-PRO-05** [comment]
I and tg are working on SPARKLER-93 branch #110. This pull request now becomes a duplicate request -- @thammegowda I think I should close this now. @chrismattmann you can review the pull request #110. But it is not complete we are stuck on multiple class loading somewhere by Felix.

**Comment by chrismattmann** [comment]
got it, closed

**Pull request closed without merging**